source/common/upstream/health_checker_impl.h

```diff
@@ -173,12 +172,24 @@ class HttpHealthCheckerImpl : public HealthCheckerImplBase {
 
     // Http::StreamCallbacks
     void onResetStream(Http::StreamResetReason reason) override;
-
-    // Network::ConnectionCallbacks
-    void onEvent(uint32_t events) override;
     void onAboveWriteBufferHighWatermark() override {}
     void onBelowWriteBufferLowWatermark() override {}
 
+    void onEvent(uint32_t events);
+
+    class ConnectionCallbackImpl : public Network::ConnectionCallbacks {
+    public:
+      ConnectionCallbackImpl(HttpActiveHealthCheckSession& parent) : parent_(parent) {}
+      // Network::ConnectionCallbacks
+      void onEvent(uint32_t events) override { parent_.onEvent(events); }
+      void onAboveWriteBufferHighWatermark() override {}
```

Comment:
Is this for the no-op functions?  I was pretty careful on coverage to cover actual lines of code.
For no-ops do we want to have the tests all call onUnusedfunction() or are coverage annotations to disregard ok?